compiler/optimizer/LocalOpts.cpp

```diff
@@ -6811,83 +6813,71 @@ int32_t TR_InvariantArgumentPreexistence::perform()
             int32_t index = symbol->getParmSymbol()->getOrdinal();
             _parmInfo[index].setNotInvariant();
             --numInvariantArgs;
-            if (trace())
+            if (enableTrace)
                traceMsg(comp(), "PREX:    Arg %d (%s) is not invariant\n", index, node->getSymbolReference()->getName(comp()->getDebug()));
             }
          }
       }
 
-
+   TR::KnownObjectTable *knot = comp()->getKnownObjectTable();
```

Comment:
need to check for null knot here if the table is disabled I believe

Reply:
The code doesn't add new entry to the table, it just queries the table with existing known object index, so if we can get a valid known object index, that means the table is not disable.

Reply:
Right but why consume the compile time traversing everything if the knot is NULL?

Reply:
The traversing is not only about getting the known object, but also about collecting arg info from parm symbol and prex arg info.